Implement UserProfilesResource, VisitorsResource and ContactsResource

dhartwich1991 opened the pull request:
Implement UserProfilesResource, VisitorsResource and ContactsResource

To review: 
- [x] @serj-lotutovici 

List of Commits in this: 

commit 0431750538dada50ec1c30cc3ada6c40908ea250
Author: Daniel Hartwich daniel_hartwich@icloud.com
Date:   Thu Dec 10 17:48:12 2015 +0100

```
Implement ContactsResource and ProfileVisitsResource
```

commit b3c9274961bcef32d35675deeb1e9eedff63a4c1
Author: Daniel Hartwich daniel_hartwich@icloud.com
Date:   Thu Dec 10 09:58:48 2015 +0100

```
Fix some checkstyle warnings
```

commit a735428ad1d972996db4cb1c4c5768bfd3950a6a
Author: Daniel Hartwich daniel_hartwich@icloud.com
Date:   Thu Dec 10 09:54:17 2015 +0100

```
add a lot of javadoc
```

serj-lotutovici commented on sample/src/main/java/com/xing/android/sdk/sample/ProfileActivity.java:
```diff
@@ -245,9 +247,22 @@ protected XingUser doInBackground(Activity... params) {
+                    @Override
+                    public void onResponse(Response<String, String> response) {
+                        if (response.isSuccess())
+                        Log.i("Update res", response.raw().toString());
```
Should be ether in the same line with the if... of surrounded by brackets.

serj-lotutovici commented on sdk/src/main/java/com/xing/android/sdk/model/ContactPaths.java:
```diff
@@ -0,0 +1,70 @@
+/**
+ * @author daniel.hartwich
+ */
+public class ContactPaths {
```
Note to self. This object requires a custom json adapter.

serj-lotutovici commented on sdk/src/main/java/com/xing/android/sdk/resources/ContactsResource.java:
```diff
@@ -22,17 +22,380 @@
+     * @param recipientId Recipient ID
+     * @param senderId Sender ID
+     */
+    public CallSpec<String, String> denyOrRevokeContactRequesdt(String recipientId, String senderId) {
```
Reques**d**t?

serj-lotutovici commented on sdk/src/main/java/com/xing/android/sdk/resources/ContactsResource.java:
```diff
@@ -22,17 +22,380 @@
+     * @param userId ID of the user whose contact path(s) are to be returned
+     * @param otherUserId ID of any other XING user
+     */
+    public CallSpec<ContactPaths, Object> getContactPath(String userId, String otherUserId) {
```
may be getter to say `getContactPaths`?

serj-lotutovici commented on sdk/src/main/java/com/xing/android/sdk/task/contact_petition/IncomingContactPetitionsTask.java:
```diff
@@ -40,7 +40,7 @@
  */
 @SuppressWarnings("unused")
-public class IncomingContactPetitionsTask extends PaginatedWithOffsetTask<List<ContactPetition>> {
+public class IncomingContactPetitionsTask extends PaginatedWithOffsetTask<List<ContactRequest>> {
```
Waiting for the day when we delete this stuff.

dhartwich1991 replied:
Me too my friend, me too :+1: :heart_eyes: :shipit:

serj-lotutovici commented:
LGTM, aside from a few formatting issues.

dhartwich1991 committed:
Implement UserProfilesResource, VisitorsResource and ContactsResource

    commit 0431750538dada50ec1c30cc3ada6c40908ea250
    Author: Daniel Hartwich <daniel_hartwich@icloud.com>
    Date:   Thu Dec 10 17:48:12 2015 +0100

        Implement ContactsResource and ProfileVisitsResource

    commit b3c9274961bcef32d35675deeb1e9eedff63a4c1
    Author: Daniel Hartwich <daniel_hartwich@icloud.com>
    Date:   Thu Dec 10 09:58:48 2015 +0100

        Fix some checkstyle warnings

    commit a735428ad1d972996db4cb1c4c5768bfd3950a6a
    Author: Daniel Hartwich <daniel_hartwich@icloud.com>
    Date:   Thu Dec 10 09:54:17 2015 +0100

        add a lot of javadoc

commit 28d10ea93e8b8c43029bea769185275b00b767c5
Merge: aa7b474 39b0c28
Author: Serj Lotutovici <slotutovici@gmail.com>
Date:   Wed Dec 9 18:22:19 2015 +0100

    Merge pull request #38 from xing/sl/more_on_callspec

    Ensure CallSpec acts as expected.

commit 39b0c2827e3dea7b4435514a48edb35e8c88790d
Author: Serj Lotutovici <serj.lotutovici@xing.com>
Date:   Wed Dec 9 17:47:45 2015 +0100

    Ensure CallSpec acts as expected.

dhartwich1991 commented:
fixed the above issues
